cluster-agent/controllers/argoproj.io/application_controller.go

```diff
@@ -149,6 +149,14 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 			applicationState.ReconciledState = reconciledState
 
+			// Update applicationState.SyncError with the ArgoCD application syncError message
+			for _, syncError := range app.Status.Conditions {
+				// Update syncError field of appliactionState only if type is SyncError
+				if syncError.Type == appv1.ApplicationConditionSyncError {
+					applicationState.SyncError = db.TruncateVarchar(syncError.Message, db.ApplicationStateSyncErrorLength)
```

Comment:
This logic should also handle the case where the ApplicationConditionSyncError is not found, in which case SyncError should be set to `""`

Reply:
yes right @jgwest  I didn't realised this 😅, thank you , I will update this